sceptre_cdk_handler/cdk_builder.py

```diff
@@ -0,0 +1,214 @@
+import logging
+import os
+import subprocess
+import sys
+from abc import ABC, abstractmethod
+from typing import Any, Optional, Dict, Type
+
+import aws_cdk
+from aws_cdk.cx_api import CloudAssembly
+from botocore.credentials import Credentials
+from cdk_bootstrapless_synthesizer import BootstraplessStackSynthesizer
+from sceptre import exceptions
+from sceptre.connection_manager import ConnectionManager
+from sceptre.exceptions import TemplateHandlerArgumentsInvalidError
+
+
+class SceptreCdkStack(aws_cdk.Stack):
+    def __init__(self, scope: aws_cdk.App, id: str, sceptre_user_data: Any, **kwargs):
+        super().__init__(scope, id, **kwargs)
+        self.sceptre_user_data = sceptre_user_data
+
+
+class CdkBuilder(ABC):
+    STACK_LOGICAL_ID = 'CDKStack'
+
+    @abstractmethod
+    def build_template(
+        self,
+        stack_class: Type[SceptreCdkStack],
+        cdk_context: Optional[dict],
+        sceptre_user_data: Any
+    ) -> dict: ...
+
+
+class BootstrappedCdkBuilder(CdkBuilder):
+
+    def __init__(
+        self,
+        logger: logging.Logger,
+        connection_manager: ConnectionManager,
+        *,
+        subprocess_run=subprocess.run,
+        app_class=aws_cdk.App,
+        environment_variables=os.environ
+    ):
+        self._logger = logger
+        self._connection_manager = connection_manager
+        self._subprocess_run = subprocess_run
+        self._app_class = app_class
+        self._environment_variables = environment_variables
+
+    def build_template(
+        self,
+        stack_class: Type[SceptreCdkStack],
+        cdk_context: Optional[dict],
+        sceptre_user_data: Any
+    ) -> dict:
+        assembly = self._synthesize(stack_class, cdk_context, sceptre_user_data)
+        self._publish(assembly)
+        template = self._get_template(assembly)
+        return template
+
+    def _synthesize(
+        self,
+        stack_class: Type[SceptreCdkStack],
+        cdk_context: Optional[dict],
+        sceptre_user_data: Any
+    ) -> CloudAssembly:
+        self._logger.debug(f'CDK synthesing CdkStack Class')
```

Comment:
couldn't this be named something different than `CdkStack` class?

Reply:
Whoops! That was a hold-over from before. I'll make this a bit better...